packages/ra-auth-google/src/getGoogleAuthProvider.tsx

```diff
@@ -22,33 +22,31 @@ import { TokenStore, localStorageTokenStore } from './tokenStore';
  *
  * @example
  * ```ts
- * const { authProvider, httpClient, gsiParams } = useGoogleAuthProvider();
+ * const { authProvider, httpClient, gsiParams } = getGoogleAuthProvider();
  * ```
  *
  * @example
  * ```ts
- * const { authProvider, httpClient, gsiParams } = useGoogleAuthProvider({
+ * const { authProvider, httpClient, gsiParams } = getGoogleAuthProvider({
  *   client_id: "my-application-client-id.apps.googleusercontent.com",
  *   context: "use",
  *   tokenStore: myTokenStore,
  * });
  * ```
  */
-export const useGoogleAuthProvider = (params?: UseGoogleAuthProviderParams) => {
+export const getGoogleAuthProvider = (params?: getGoogleAuthProviderParams) => {
```

Comment:
As we're going for breaking changes anyway, I think this method is either wrongly named or shouldn't exist at all. We already have `googleAuthProvider` and `googleHttpClient` and that's probably enough. @slax57, @fzaninotto?